--- src-original/setup.py	2025-07-07 12:06:03.000000000 +0000
+++ src/setup.py	2025-11-06 13:14:14.869402601 +0000
@@ -18,6 +18,21 @@
 import shutil
 import sys
 
+# Chaquopy: complex.h functions were introduced between API levels 23 and 26, so make
+# Cython use its own implementations instead.
+#
+# This isn't a problem with Cython's C++ mode, e.g. in qutip, presumably because libcxx
+# contains its own implementations of complex number functions in order to be compatible
+# with the minimum API level supported by the NDK. In fact, setting CYTHON_CCOMPLEX=0
+# for such a module causes Cython to generate code that passes Cython's own complex type
+# to C++ standard library functions, which obviously fails to compile. So that's why we
+# can't add this to the default CFLAGS.
+os.environ["CFLAGS"] += " -DCYTHON_CCOMPLEX=0"
+
+# Chaquopy: prevent the compiled parts of NumPy from being imported.
+import builtins
+builtins.__NUMPY_SETUP__ = True
+
 import numpy as np
 from packaging.version import parse
 
@@ -259,7 +274,7 @@
 
 
 NUMPY_INCLUDES = sorted(
-    {np.get_include(), pjoin(dirname(inspect.getfile(np.core)), "include")}
+    {np.get_include()}  # Chaquopy: removed pjoin(dirname(inspect.getfile(np.core)), "include")
 )
 NUMPY_MATH_LIBS = {
     "include_dirs": [np.get_include()],
