--- src-original/setup.py	2019-12-05 17:41:56.000000000 +0000
+++ src/setup.py	2020-01-29 19:56:11.705956312 +0000
@@ -17,7 +17,7 @@
 try:
     # SM_FORCE_C is a testing shim to force setup to use C source files
     FORCE_C = int(os.environ.get('SM_FORCE_C', 0))
-    if FORCE_C:
+    if True or FORCE_C:  # Chaquopy: Cython version 0.29.14 gives errors.
         raise ImportError('Force import error for testing')
     from Cython import Tempita
     from Cython.Build import cythonize
@@ -199,14 +199,8 @@
         build_ext.build_extensions(self)
 
     def _update_extensions(self):
-        import numpy
-        from numpy.distutils.misc_util import get_info
-
-        numpy_includes = [numpy.get_include()]
-        extra_incl = pkg_resources.resource_filename('numpy', 'core/include')
-        numpy_includes += [extra_incl]
-        numpy_includes = list(set(numpy_includes))
-        numpy_math_libs = get_info('npymath')
+        # Chaquopy: removed numpy import
+        numpy_includes = []
 
         for extension in self.extensions:
             if not hasattr(extension, 'include_dirs'):
@@ -214,9 +208,10 @@
             extension.include_dirs = list(set(extension.include_dirs +
                                               numpy_includes))
             if extension.name in EXT_REQUIRES_NUMPY_MATH_LIBS:
-                extension.include_dirs += numpy_math_libs['include_dirs']
-                extension.libraries += numpy_math_libs['libraries']
-                extension.library_dirs += numpy_math_libs['library_dirs']
+                # Chaquopy: replaced use of numpy.distutils.misc_util.get_info
+                extension.include_dirs += [os.path.abspath("../requirements/numpy/core/include")]
+                extension.libraries += ["npymath"]
+                extension.library_dirs += [os.path.abspath("../requirements/numpy/core/lib")]
 
 
 cmdclass = versioneer.get_cmdclass()
@@ -341,7 +336,7 @@
       package_data=package_data,
       distclass=BinaryDistribution,
       include_package_data=False,  # True will install all files in repo
-      setup_requires=SETUP_REQUIRES,
+      # Chaquopy: removed setup_requires=SETUP_REQUIRES,
       install_requires=INSTALL_REQUIRES,
       extras_require=EXTRAS_REQUIRE,
       zip_safe=False
