diff -ru src-original/buildutils/detect.py src/buildutils/detect.py
--- src-original/buildutils/detect.py	2018-02-10 13:02:35.000000000 +0000
+++ src/buildutils/detect.py	2018-04-20 18:03:24.807279447 +0000
@@ -17,6 +17,7 @@
 import logging
 import platform
 from distutils import ccompiler
+from distutils.sysconfig import customize_compiler  # Chaquopy added
 from subprocess import Popen, PIPE
 
 from .misc import get_compiler, get_output_error
@@ -108,12 +109,22 @@
     # check if we need to link against Realtime Extensions library
     if sys.platform.startswith('linux'):
         cc = ccompiler.new_compiler(compiler=compiler)
+        customize_compiler(cc)  # Chaquopy added
         cc.output_dir = basedir
         if not cc.has_function('timer_create'):
             compiler_attrs['libraries'].append('rt')
     
     cc = get_compiler(compiler=compiler, **compiler_attrs)
     efile = test_compilation(cfile, compiler=cc, **compiler_attrs)
+
+    # Chaquopy: we can't run the test program when cross-compiling, so do this instead.
+    import re
+    for filename in os.listdir("../requirements"):
+        match = re.search(r"chaquopy-libzmq-(.*).dist-info", filename)
+        if match:
+            return {"vers": tuple(int(x) for x in match.group(1).split("."))}
+    raise Exception("Failed to find chaquopy-libzmq version")
+
     patch_lib_paths(efile, cc.library_dirs)
     
     rc, so, se = get_output_error([efile])
diff -ru src-original/setup.py src/setup.py
--- src-original/setup.py	2018-02-10 13:02:35.000000000 +0000
+++ src/setup.py	2018-04-20 17:51:47.093598878 +0000
@@ -85,6 +85,9 @@
 # allow `--zmq=foo` to be passed at any point,
 # but always assign it to configure
 
+# Chaqupy added
+sys.argv.append("--zmq=../requirements/chaquopy")
+
 configure_idx = -1
 fetch_idx = -1
 for idx, arg in enumerate(list(sys.argv)):
