diff --git a/Cargo.toml b/Cargo.toml
index 4dbfef9..bb4f88a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,6 +18,9 @@ python = [
 pyo3 = { version = "0.22.2", default-features = false, features = [
     "extension-module",
     "macros",
+    # Chaquopy: Add abi3-py312 feature, which depends on the Python version being built for.
+    # In this case, it's for Python 3.12.
+    "abi3-py312",
 ], optional = true }
 
 # tiktoken dependencies
