--- src-original/setupext.py	2022-09-15 23:26:26.000000000 +0000
+++ src/setupext.py	2022-10-04 20:30:48.569071439 +0000
@@ -199,7 +199,7 @@
     'backend': config.get('rc_options', 'backend', fallback=None),
     'system_freetype': config.getboolean(
         'libs', 'system_freetype',
-        fallback=sys.platform.startswith(('aix', 'os400'))
+        fallback=True  # Chaquopy: was sys.platform.startswith(('aix', 'os400'))
     ),
     'system_qhull': config.getboolean(
         'libs', 'system_qhull', fallback=sys.platform.startswith('os400')
@@ -493,6 +493,10 @@
 
 
 def add_numpy_flags(ext):
+    # Chaquopy
+    import builtins
+    builtins.__NUMPY_SETUP__ = True  # Prevent NumPy runtime from being imported.
+
     import numpy as np
     ext.include_dirs.append(np.get_include())
     ext.define_macros.extend([
