--- src-original/setupext.py	2019-11-21 22:47:05.000000000 +0000
+++ src/setupext.py	2020-01-26 19:16:23.926678121 +0000
@@ -604,8 +604,7 @@
     name = "numpy"
 
     def add_flags(self, ext):
-        import numpy as np
-        ext.include_dirs.append(np.get_include())
+        ext.include_dirs.append(os.path.abspath("../requirements/numpy/core/include"))  # Chaquopy
         ext.define_macros.extend([
             # Ensure that PY_ARRAY_UNIQUE_SYMBOL is uniquely defined for each
             # extension.
@@ -617,7 +616,7 @@
         ])
 
     def get_setup_requires(self):
-        return ['numpy>=1.11']
+        return []  # Chaquopy: was ['numpy>=1.11']
 
     def get_install_requires(self):
         return ['numpy>=1.11']
@@ -679,14 +678,7 @@
                 0, os.path.join(src_path, 'objs', '.libs', libfreetype))
             ext.define_macros.append(('FREETYPE_BUILD_TYPE', 'local'))
         else:
-            pkg_config.setup_extension(
-                # FreeType 2.3 has libtool version 9.11.3 as can be checked
-                # from the tarball.  For FreeType>=2.4, there is a conversion
-                # table in docs/VERSIONS.txt in the FreeType source tree.
-                ext, 'freetype2',
-                atleast_version='9.11.3',
-                alt_exec=['freetype-config'],
-                default_libraries=['freetype', 'z'])
+            ext.libraries.append("freetype")  # Chaquopy
             ext.define_macros.append(('FREETYPE_BUILD_TYPE', 'system'))
 
     def do_custom_build(self):
