--- src-original/setupext.py	2022-09-15 23:26:26.000000000 +0000
+++ src/setupext.py	2022-10-04 20:30:48.569071439 +0000
@@ -199,7 +199,7 @@
     'backend': config.get('rc_options', 'backend', fallback=None),
     'system_freetype': config.getboolean(
         'libs', 'system_freetype',
-        fallback=sys.platform.startswith(('aix', 'os400'))
+        fallback=True  # Chaquopy: was sys.platform.startswith(('aix', 'os400'))
     ),
     'system_qhull': config.getboolean(
         'libs', 'system_qhull', fallback=sys.platform.startswith('os400')
@@ -493,6 +493,10 @@
 
 
 def add_numpy_flags(ext):
+    # Chaquopy
+    import builtins
+    builtins.__NUMPY_SETUP__ = True  # Prevent NumPy runtime from being imported.
+
     import numpy as np
     ext.include_dirs.append(np.get_include())
     ext.define_macros.extend([
--- src-original/setup.py	2022-09-15 23:26:26.000000000 +0000
+++ src/setup.py	2022-10-04 21:05:37.058026731 +0000
@@ -302,7 +302,7 @@
     python_requires='>={}'.format('.'.join(str(n) for n in py_min_version)),
     setup_requires=[
         "certifi>=2020.06.20",
-        "numpy>=1.19",
+        # Chaquopy: removed "numpy>=1.19",
         "setuptools_scm>=7",
     ],
     install_requires=[
