--- src-original/setupext.py	2019-03-01 03:33:38.000000000 +0000
+++ src/setupext.py	2019-03-14 17:40:25.157405925 +0000
@@ -243,6 +243,9 @@
     """
     Returns a list of standard base directories on this platform.
     """
+    # Chaquopy
+    return [os.path.abspath("../requirements/chaquopy")]
+
     if options['basedirlist']:
         return options['basedirlist']
 
@@ -960,6 +963,7 @@
 
     @staticmethod
     def include_dirs_hook():
+        return [os.path.abspath("../requirements/numpy/core/include")]  # Chaquopy
         if hasattr(builtins, '__NUMPY_SETUP__'):
             del builtins.__NUMPY_SETUP__
         import numpy
@@ -1004,7 +1008,7 @@
         ext.define_macros.append(('__STDC_FORMAT_MACROS', 1))
 
     def get_setup_requires(self):
-        return ['numpy>=1.10.0']
+        return []  # Chaquopy: removed numpy
 
     def get_install_requires(self):
         return ['numpy>=1.10.0']
@@ -1057,7 +1061,7 @@
         if options.get('local_freetype'):
             return "Using local version for testing"
 
-        if sys.platform == 'win32':
+        if True:  # Chaquopy: was `if sys.platform == 'win32':`
             try:
                 check_include_file(get_include_dirs(), 'ft2build.h', 'freetype')
             except CheckFailed:
