--- src-original/psutil/_pslinux.py	2025-10-23 21:01:06.000000000 +0000
+++ src/psutil/_pslinux.py	2025-11-05 18:37:13.930999156 +0000
@@ -1248,6 +1248,10 @@
     """Return mounted disk partitions as a list of namedtuples."""
     fstypes = set()
     procfs_path = get_procfs_path()
+
+    # Chaquopy: /proc/filesystems is blocked by SELinux on API level 28.
+    all = True
+
     if not all:
         with open_text(f"{procfs_path}/filesystems") as f:
             for line in f:
