diff -ur src-original/numpy/distutils/system_info.py src/numpy/distutils/system_info.py
--- src-original/numpy/distutils/system_info.py 2018-03-12 09:49:52.000000000 +0000
+++ src/numpy/distutils/system_info.py  2018-04-17 12:42:12.265904840 +0000
@@ -1876,7 +1876,8 @@
         c = customized_ccompiler()

         tmpdir = tempfile.mkdtemp()
-        s = """void zungqr();
+        # Chaquopy: added missing underscore (https://github.com/numpy/numpy/pull/10916)
+        s = """void zungqr_();
         int main(int argc, const char *argv[])
         {
             zungqr_();
diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py       2018-04-15 20:25:55.769497865 +0000
+++ src/setup.py        2018-04-15 19:32:59.070017179 +0000
@@ -39,6 +39,10 @@
     import __builtin__ as builtins


+# Chaquopy added
+os.environ["OPENBLAS"] = os.path.abspath("../requirements/chaquopy/lib")
+
+
 CLASSIFIERS = """\
 Development Status :: 5 - Production/Stable
 Intended Audience :: Science/Research
