--- src-original/setup.py	2021-08-24 11:50:48.172484900 +0000
+++ src/setup.py	2022-07-11 22:15:55.179880743 +0000
@@ -11,6 +11,11 @@
 from distutils.command.build_py import build_py
 from distutils.command.sdist import sdist
 from distutils.errors import CompileError, LinkError
+
+# Chaquopy
+sys.path.insert(0, os.path.abspath("../requirements"))  # For numpy.distutils
+builtins.__NUMPY_SETUP__ = True  # Prevent the compiled parts from being imported.
+
 from numpy.distutils.command.build_ext import build_ext
 
 
