--- src-original/src/rust/src/lib.rs
+++ src/src/rust/src/lib.rs
@@ -62,8 +62,8 @@
     // are ugly legacy, but we aren't going to get rid of them
     // any time soon.

-    let load_legacy = !cfg!(CRYPTOGRAPHY_BUILD_OPENSSL_NO_LEGACY)
-        && !env::var("CRYPTOGRAPHY_OPENSSL_NO_LEGACY").is_ok_and(|v| !v.is_empty() && v != "0");
+    // Chaquopy: the legacy provider is not available.
+    let load_legacy = false;

     let legacy = if load_legacy {
         let legacy_result = provider::Provider::load(None, "legacy");
