diff -ur src-original/lib/Crypto/Util/_raw_api.py src/lib/Crypto/Util/_raw_api.py
--- src-original/lib/Crypto/Util/_raw_api.py    2018-08-04 22:21:17.000000000 +0000
+++ src/lib/Crypto/Util/_raw_api.py     2019-08-24 18:50:56.520037232 +0000
@@ -255,6 +255,18 @@
                             cdecl)
         except OSError as exp:
             attempts.append("Trying '%s': %s" % (filename, str(exp)))
+
+    # This technique will work both before and after the importer redesign in Chaquopy 6.3.
+    import pkgutil
+    import Crypto
+    for entry in Crypto.__path__:
+        importer = pkgutil.get_importer(entry)
+        try:
+            filename = importer.extract_if_changed(name.replace(".", "/") + ".so")
+            return load_lib(filename, cdecl)
+        except KeyError:
+            attempts.append("Trying importer for '%s'" % entry)
+
     raise OSError("Cannot load native module '%s': %s" % (name, ", ".join(attempts)))

 def expect_byte_string(data):
diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2018-08-13 19:37:53.000000000 +0000
+++ src/setup.py	2018-09-28 18:49:58.518694480 +0000
@@ -537,43 +537,11 @@
     "Crypto.Signature",
     "Crypto.Util",
     "Crypto.Math",
-    "Crypto.SelfTest",
-    "Crypto.SelfTest.Cipher",
-    "Crypto.SelfTest.Hash",
-    "Crypto.SelfTest.IO",
-    "Crypto.SelfTest.Protocol",
-    "Crypto.SelfTest.PublicKey",
-    "Crypto.SelfTest.Random",
-    "Crypto.SelfTest.Signature",
-    "Crypto.SelfTest.Util",
-    "Crypto.SelfTest.Math",
+    # Chaquopy: removed large package Crypto.SelfTest
 ]
 package_dir = { "Crypto": "lib/Crypto" }
 package_data = {
-    "Crypto.SelfTest.Cipher" : [
-        "test_vectors/AES/*.rsp",
-        "test_vectors/TDES/*.rsp",
-        "test_vectors/wycheproof/*.json",
-    ],
-    "Crypto.SelfTest.Hash" : [
-        "test_vectors/SHA1/*.rsp",
-        "test_vectors/SHA2/*.rsp",
-        "test_vectors/SHA3/*.txt",
-        "test_vectors/keccak/*.txt",
-        "test_vectors/BLAKE2s/*.txt",
-        "test_vectors/BLAKE2b/*.txt",
-        "test_vectors/wycheproof/*.json",
-    ],
-    "Crypto.SelfTest.Signature" : [
-        "test_vectors/DSA/*.*",
-        "test_vectors/ECDSA/*.*",
-        "test_vectors/PKCS1-v1.5/*.*",
-        "test_vectors/PKCS1-PSS/*.*",
-        "test_vectors/wycheproof/*.json",
-    ],
-    "Crypto.SelfTest.PublicKey" : [
-        "test_vectors/ECC/*.*",
-    ],
+    # Chaquopy: removed large package Crypto.SelfTest
 }
 
 system_bits = 8 * struct.calcsize("P")
