diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2017-06-01 07:25:11.000000000 +0000
+++ src/setup.py	2020-01-04 16:59:25.190549111 +0000
@@ -97,7 +97,7 @@
             readme = f.read()
 
         include_dirs = [
-            get_python_inc(plat_specific=True),
+            # Chaquopy removed: get_python_inc(plat_specific=True),
             os.path.join(root, 'murmurhash', 'include')]
 
         if (ccompiler.new_compiler().compiler_type == 'msvc'
