diff --git a/src/netinterfaces.cpp b/src/netinterfaces.cpp
index 860670f..7afd105 100644
--- a/src/netinterfaces.cpp
+++ b/src/netinterfaces.cpp
@@ -114,7 +114,7 @@ std::vector<lsl::netif> lsl::get_local_interfaces() {
 }
 #else
 
-std::vector<lsl::netif> get_interface_addresses() {
+std::vector<lsl::netif> lsl::get_local_interfaces() {
 	LOG_F(WARNING, "No implementation to enumerate network interfaces found.");
 	return std::vector<lsl::netif>();
 }
