--- src-original/setup.py	2020-05-10 14:00:25.000000000 +0000
+++ src/setup.py	2021-02-27 16:03:58.134283521 +0000
@@ -27,7 +27,7 @@
 else:
     # Optimized version requires SSE2 extensions.  They have been around since
     # 2001 so we try to compile it on every recent-ish x86.
-    optimized = platform.machine() in ("i686", "x86", "x86_64", "AMD64")
+    optimized = os.environ["CHAQUOPY_ABI"] in ("i686", "x86", "x86_64", "AMD64")  # Was platform.machine()
 
 CFFI_MODULES = ["src/argon2/_ffi_build.py:ffi"]
 lib_base = os.path.join("extras", "libargon2", "src")
