diff -ur src-original/scipy/__init__.py src/scipy/__init__.py
--- src-original/scipy/__init__.py	2018-03-17 21:04:47.000000000 +0000
+++ src/scipy/__init__.py	2018-10-16 22:06:56.116131775 +0000
@@ -58,32 +58,6 @@
 
 __all__ = ['test']
 
-from numpy import show_config as show_numpy_config
-if show_numpy_config is None:
-    raise ImportError(
-        "Cannot import scipy when running from numpy source directory.")
-from numpy import __version__ as __numpy_version__
-
-# Import numpy symbols to scipy name space
-import numpy as _num
-linalg = None
-from numpy import *
-from numpy.random import rand, randn
-from numpy.fft import fft, ifft
-from numpy.lib.scimath import *
-
-# Allow distributors to run custom init code
-from . import _distributor_init
-
-__all__ += _num.__all__
-__all__ += ['randn', 'rand', 'fft', 'ifft']
-
-del _num
-# Remove the linalg imported from numpy so that the scipy.linalg package can be
-# imported.
-del linalg
-__all__.remove('linalg')
-
 # We first need to detect if we're being called as part of the scipy
 # setup procedure itself in a reliable manner.
 try:
@@ -96,7 +70,40 @@
     import sys as _sys
     _sys.stderr.write('Running from scipy source directory.\n')
     del _sys
+
+    # Chaquopy: the module namespace was originally empty in this mode, but the following
+    # things are neded by the scikit-learn build.
+    from scipy.version import version as __version__
+
 else:
+    # Chaquopy: BEGIN section moved from top level
+    from numpy import show_config as show_numpy_config
+    if show_numpy_config is None:
+        raise ImportError(
+            "Cannot import scipy when running from numpy source directory.")
+    from numpy import __version__ as __numpy_version__
+
+    # Import numpy symbols to scipy name space
+    import numpy as _num
+    linalg = None
+    from numpy import *
+    from numpy.random import rand, randn
+    from numpy.fft import fft, ifft
+    from numpy.lib.scimath import *
+
+    # Allow distributors to run custom init code
+    from . import _distributor_init
+
+    __all__ += _num.__all__
+    __all__ += ['randn', 'rand', 'fft', 'ifft']
+
+    del _num
+    # Remove the linalg imported from numpy so that the scipy.linalg package can be
+    # imported.
+    del linalg
+    __all__.remove('linalg')
+    # Chaquopy: END section moved from top level
+
     try:
         from scipy.__config__ import show as show_config
     except ImportError:
diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2018-03-24 03:12:35.000000000 +0000
+++ src/setup.py	2018-10-16 22:36:07.845070738 +0000
@@ -33,6 +33,20 @@
 else:
     import builtins
 
+
+# Chaquopy: arm64-v8a requires -rpath-link, otherwise it can't find libgfortran.so.3 when
+# linking the test program. This is in the toolchain at aarch64-linux-android/lib64, but
+# symlinking it into aarch64-linux-android/lib or sysroot/usr/lib made no difference.
+with open(os.path.join(os.path.dirname(__file__), "site.cfg"), "w") as site_file:
+    site_file.write("[openblas]\n"
+                    "library_dirs = {reqs_dir}/lib\n"
+                    "include_dirs = {reqs_dir}/include\n"
+                    "extra_link_args = -Wl,-rpath-link,{reqs_dir}/lib\n"
+                    .format(reqs_dir=os.path.abspath("../requirements/chaquopy")))
+sys.path.append(os.path.abspath("../requirements"))  # For numpy.distutils
+builtins.__NUMPY_SETUP__ = True  # Prevent the rest of NumPy from being imported.
+
+
 CLASSIFIERS = """\
 Development Status :: 5 - Production/Stable
 Intended Audience :: Science/Research
@@ -377,7 +391,7 @@
         classifiers=[_f for _f in CLASSIFIERS.split('\n') if _f],
         platforms=["Windows", "Linux", "Solaris", "Mac OS-X", "Unix"],
         test_suite='nose.collector',
-        setup_requires=build_requires,
+        # Chaquopy: removed setup_requires=build_requires
         install_requires=build_requires,
         python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',
     )
