diff -Naur src-original/scipy/__init__.py src/scipy/__init__.py
--- src-original/scipy/__init__.py	2023-08-31 12:17:17.197651190 +0200
+++ src/scipy/__init__.py	2023-08-31 12:17:39.438111356 +0200
@@ -58,50 +58,54 @@
 """
 __all__ = ['test']
 
-from numpy import show_config as show_numpy_config
-if show_numpy_config is None:
-    raise ImportError(
-        "Cannot import SciPy when running from NumPy source directory.")
-from numpy import __version__ as __numpy_version__
-
-# Import numpy symbols to scipy name space (DEPRECATED)
-from ._lib.deprecation import _deprecated
-import numpy as _num
-linalg = None
-_msg = ('scipy.{0} is deprecated and will be removed in SciPy 2.0.0, '
-        'use numpy.{0} instead')
-# deprecate callable objects, skipping classes
-for _key in _num.__all__:
-    _fun = getattr(_num, _key)
-    if callable(_fun) and not isinstance(_fun, type):
-        _fun = _deprecated(_msg.format(_key))(_fun)
-    globals()[_key] = _fun
-from numpy.random import rand, randn
-_msg = ('scipy.{0} is deprecated and will be removed in SciPy 2.0.0, '
-        'use numpy.random.{0} instead')
-rand = _deprecated(_msg.format('rand'))(rand)
-randn = _deprecated(_msg.format('randn'))(randn)
-# fft is especially problematic, so was removed in SciPy 1.6.0
-from numpy.fft import ifft
-ifft = _deprecated('scipy.ifft is deprecated and will be removed in SciPy '
-                   '2.0.0, use scipy.fft.ifft instead')(ifft)
-import numpy.lib.scimath as _sci
-_msg = ('scipy.{0} is deprecated and will be removed in SciPy 2.0.0, '
-        'use numpy.lib.scimath.{0} instead')
-for _key in _sci.__all__:
-    _fun = getattr(_sci, _key)
-    if callable(_fun):
-        _fun = _deprecated(_msg.format(_key))(_fun)
-    globals()[_key] = _fun
-
-__all__ += _num.__all__
-__all__ += ['randn', 'rand', 'ifft']
-
-del _num
-# Remove the linalg imported from NumPy so that the scipy.linalg package can be
-# imported.
-del linalg
-__all__.remove('linalg')
+# Chaquopy: don't import numpy during build.
+try:
+    __SCIPY_SETUP__
+except NameError:
+    from numpy import show_config as show_numpy_config
+    if show_numpy_config is None:
+        raise ImportError(
+            "Cannot import SciPy when running from NumPy source directory.")
+    from numpy import __version__ as __numpy_version__
+
+    # Import numpy symbols to scipy name space (DEPRECATED)
+    from ._lib.deprecation import _deprecated
+    import numpy as _num
+    linalg = None
+    _msg = ('scipy.{0} is deprecated and will be removed in SciPy 2.0.0, '
+            'use numpy.{0} instead')
+    # deprecate callable objects, skipping classes
+    for _key in _num.__all__:
+        _fun = getattr(_num, _key)
+        if callable(_fun) and not isinstance(_fun, type):
+            _fun = _deprecated(_msg.format(_key))(_fun)
+        globals()[_key] = _fun
+    from numpy.random import rand, randn
+    _msg = ('scipy.{0} is deprecated and will be removed in SciPy 2.0.0, '
+            'use numpy.random.{0} instead')
+    rand = _deprecated(_msg.format('rand'))(rand)
+    randn = _deprecated(_msg.format('randn'))(randn)
+    # fft is especially problematic, so was removed in SciPy 1.6.0
+    from numpy.fft import ifft
+    ifft = _deprecated('scipy.ifft is deprecated and will be removed in SciPy '
+                    '2.0.0, use scipy.fft.ifft instead')(ifft)
+    import numpy.lib.scimath as _sci
+    _msg = ('scipy.{0} is deprecated and will be removed in SciPy 2.0.0, '
+            'use numpy.lib.scimath.{0} instead')
+    for _key in _sci.__all__:
+        _fun = getattr(_sci, _key)
+        if callable(_fun):
+            _fun = _deprecated(_msg.format(_key))(_fun)
+        globals()[_key] = _fun
+
+    __all__ += _num.__all__
+    __all__ += ['randn', 'rand', 'ifft']
+
+    del _num
+    # Remove the linalg imported from NumPy so that the scipy.linalg package can be
+    # imported.
+    del linalg
+    __all__.remove('linalg')
 
 # We first need to detect if we're being called as part of the SciPy
 # setup procedure itself in a reliable manner.
@@ -115,6 +119,11 @@
     import sys as _sys
     _sys.stderr.write('Running from SciPy source directory.\n')
     del _sys
+
+    # Chaquopy: the module namespace was originally empty in this mode, but the following
+    # things are neded by the scikit-learn build.
+    from scipy.version import version as __version__
+
 else:
     try:
         from scipy.__config__ import show as show_config
--- src-original/setup.py	2021-04-18 14:28:08.000000000 +0000
+++ src/setup.py	2023-09-09 15:45:15.243083019 +0000
@@ -33,6 +33,20 @@
 import builtins
 
 
+
+# Chaquopy: an OpenBLAS test executable is built without using LDFLAGS, so we have to add
+# --rpath-link here: see note in build-wheel.py.
+with open(os.path.join(os.path.dirname(__file__), "site.cfg"), "w") as site_file:
+    site_file.write("[openblas]\n"
+                    "library_dirs = {reqs_dir}/lib\n"
+                    "include_dirs = {reqs_dir}/include\n"
+                    "extra_link_args = -Wl,-rpath-link,{reqs_dir}/lib\n"
+                    .format(reqs_dir=os.path.abspath("../requirements/chaquopy")))
+
+# Chaquopy: prevent the compiled parts NumPy from being imported
+builtins.__NUMPY_SETUP__ = True
+
+
 CLASSIFIERS = """\
 Development Status :: 5 - Production/Stable
 Intended Audience :: Science/Research
diff -Naur src-original/scipy/special/utils/makenpz.py src/scipy/special/utils/makenpz.py
--- src-original/scipy/special/utils/makenpz.py	2023-08-31 12:17:17.365654666 +0200
+++ src/scipy/special/utils/makenpz.py	2023-08-31 12:17:39.442111438 +0200
@@ -6,7 +6,8 @@
 """
 
 import os
-import numpy as np
+# Chaquopy: removed `import numpy`: we will "Skip rebuilding if no sources" below, so it's not
+# needed.
 import argparse
 
 from distutils.util import newer
