diff -ur src-original/config.sub src/config.sub
--- src-original/config.sub	2006-11-08 07:44:58.000000000 +0000
+++ src/config.sub	2018-11-02 18:27:35.258039102 +0000
@@ -286,7 +286,7 @@
 	| v850 | v850e \
 	| we32k \
 	| x86 | xc16x | xscale | xscalee[bl] | xstormy16 | xtensa \
-	| z8k)
+	| z8k | aarch64)  # Chaquopy: added aarch64.
 		basic_machine=$basic_machine-unknown
 		;;
 	m6811 | m68hc11 | m6812 | m68hc12)
@@ -371,7 +371,7 @@
 	| x86-* | x86_64-* | xc16x-* | xps100-* | xscale-* | xscalee[bl]-* \
 	| xstormy16-* | xtensa-* \
 	| ymp-* \
-	| z8k-*)
+	| z8k-* | aarch64-*)  # Chaquopy: added aarch64.
 		;;
 	# Recognize the various machine names and aliases which stand
 	# for a CPU type and a company and sometimes even an OS.
@@ -1219,7 +1219,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers*)
+              | -skyos* | -haiku* | -rdos* | -toppers* | -android*)  # Chaquopy: added Android.
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
