diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py       2018-06-08 01:12:35.000000000 +0000
+++ src/setup.py        2018-07-16 21:45:47.785471181 +0000
@@ -13,7 +13,12 @@
     def run(self):
         config_h = '_posixsubprocess_config.h'
         status = 'config.status'
-        configure = './configure'
+
+        # Chaquopy: added `--host`
+        cc = os.environ["CC"]
+        assert cc.endswith("-gcc"), cc
+        configure = './configure --host=' + cc[:-4]
+
         # Here we implement time stamp checking logic like make, manually...
         if os.path.exists(config_h) and os.path.exists(status):
             status_mtime = os.stat(status)[stat.ST_MTIME]
