--- src-original/setup.py	2021-11-02 19:17:42.000000000 +0000
+++ src/setup.py	2023-06-03 11:19:23.806406705 +0000
@@ -26,6 +26,10 @@
 
     @property
     def include_dirs(self):
+        # Chaquopy
+        import builtins
+        builtins.__NUMPY_SETUP__ = True  # Prevent the compiled parts from being imported.
+
         from numpy import get_include
         return self._include_dirs + [get_include()]
 
