--- src-original/setup.py	2022-07-01 13:14:51.000000000 +0000
+++ src/setup.py	2022-10-04 22:09:03.025799341 +0000
@@ -333,7 +333,7 @@
     )
 
     def initialize_options(self):
-        self.disable_platform_guessing = None
+        self.disable_platform_guessing = True  # Chaquopy: changed from None to True
         self.add_imaging_libs = ""
         build_ext.initialize_options(self)
         for x in self.feature:
@@ -636,8 +637,8 @@
 
         if feature.want("zlib"):
             _dbg("Looking for zlib")
-            if _find_include_file(self, "zlib.h"):
-                if _find_library_file(self, "z"):
+            if True or _find_include_file(self, "zlib.h"):  # Chaquopy: libz is always
+                if True or _find_library_file(self, "z"):   # available.
                     feature.zlib = "z"
                 elif sys.platform == "win32" and _find_library_file(self, "zlib"):
                     feature.zlib = "zlib"  # alternative name
