--- aaa/setup.py	2024-10-15 05:58:32.000000000 +0000
+++ src/setup.py	2024-10-16 22:01:04.906473396 +0000
@@ -355,7 +355,8 @@
         return True if value in configuration.get(option, []) else None
 
     def initialize_options(self) -> None:
-        self.disable_platform_guessing = self.check_configuration(
+        # Chaquopy: always disable platform guessing.
+        self.disable_platform_guessing = True or self.check_configuration(
             "platform-guessing", "disable"
         )
         self.add_imaging_libs = ""
@@ -685,8 +686,9 @@
 
         if feature.want("zlib"):
             _dbg("Looking for zlib")
-            if _find_include_file(self, "zlib.h"):
-                if _find_library_file(self, "z"):
+            # Chaquopy: zlib is always available.
+            if True or _find_include_file(self, "zlib.h"):
+                if True or _find_library_file(self, "z"):
                     feature.set("zlib", "z")
                 elif sys.platform == "win32" and _find_library_file(self, "zlib"):
                     feature.set("zlib", "zlib")  # alternative name
