diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2018-07-01 19:02:58.000000000 +0000
+++ src/setup.py	2018-10-17 12:15:58.387726061 +0000
@@ -194,7 +194,7 @@
     ]
 
     def initialize_options(self):
-        self.disable_platform_guessing = None
+        self.disable_platform_guessing = True  # Chaquopy: set to True
         build_ext.initialize_options(self)
         for x in self.feature:
             setattr(self, 'disable_%s' % x, None)
@@ -279,7 +279,7 @@
                     _add_directory(library_dirs, d)
 
         prefix = sysconfig.get_config_var("prefix")
-        if prefix:
+        if prefix and False:  # Chaquopy: disabled
             _add_directory(library_dirs, os.path.join(prefix, "lib"))
             _add_directory(include_dirs, os.path.join(prefix, "include"))
 
@@ -435,6 +435,12 @@
                 _add_directory(include_dirs,
                                os.path.join(best_path, 'include'))
 
+        # Chaquopy: added
+        sysroot_dir = os.path.abspath(os.path.join(os.path.dirname(os.environ["CC"]),
+                                                   "../sysroot/usr"))
+        _add_directory(library_dirs, os.path.join(sysroot_dir, "lib"))
+        _add_directory(include_dirs, os.path.join(sysroot_dir, "include"))
+
         #
         # insert new dirs *before* default libs, to avoid conflicts
         # between Python PYD stub libs and real libraries
@@ -591,7 +597,7 @@
         for src_file in _LIB_IMAGING:
             files.append(os.path.join("src/libImaging", src_file + ".c"))
 
-        libs = []
+        libs = ["m"]  # Chaquopy: added "m"
         defs = []
         if feature.jpeg:
             libs.append(feature.jpeg)
@@ -663,7 +669,8 @@
                               include_dirs=['src/Tk'],
                               libraries=tk_libs))
 
-        exts.append(Extension("PIL._imagingmath", ["src/_imagingmath.c"]))
+        exts.append(Extension("PIL._imagingmath", ["src/_imagingmath.c"],
+                              libraries=["m"]))  # Chaquopy: added
         exts.append(Extension("PIL._imagingmorph", ["src/_imagingmorph.c"]))
 
         self.extensions[:] = exts
