--- src-original/deps/c-ares/src/lib/ares_init.c	2025-09-09 11:44:31.000000000 +0000
+++ src/deps/c-ares/src/lib/ares_init.c	2025-11-06 12:30:24.226925534 +0000
@@ -155,7 +155,11 @@
     }
 
     addr.family            = AF_INET;
-    addr.addr.addr4.s_addr = htonl(INADDR_LOOPBACK);
+
+    // Chaquopy: changed INADDR_LOOPBACK to the Google public DNS server 8.8.8.8.
+    // ares_android.c has support for detecting DNS servers on Android, but it requires
+    // manual initialization with the Java ConnectivityManager.
+    addr.addr.addr4.s_addr = htonl(0x08080808);
 
     rc = ares_sconfig_append(channel, &sconfig, &addr, 0, 0, NULL);
     if (rc != ARES_SUCCESS) {
--- src-original/setup_cares.py	2025-09-09 11:44:30.000000000 +0000
+++ src/setup_cares.py	2025-11-06 12:36:06.987403774 +0000
@@ -123,7 +123,8 @@
             self.compiler.define_macro('_FILE_OFFSET_BITS', 64)
         if sys.platform.startswith('linux'):
             # Check if it's actually Android
-            if os.environ.get('ANDROID_ROOT') and os.environ.get('ANDROID_DATA'):
+            # Chaquopy: added True
+            if True or os.environ.get('ANDROID_ROOT') and os.environ.get('ANDROID_DATA'):
                 self.add_include_dir(os.path.join(self.build_config_dir, 'config_android'))
             else:
                 self.add_include_dir(os.path.join(self.build_config_dir, 'config_linux'))
