diff -ur src-original/configure src/configure
--- src-original/configure	2013-10-14 21:38:10.000000000 +0000
+++ src/configure	2018-12-29 16:10:43.497124049 +0000
@@ -3832,7 +3832,8 @@
   $as_echo_n "(cached) " >&6
 else
   if test "$cross_compiling" = yes; then :
-  ac_cv_func_malloc_0_nonnull=no
+  # Chaquopy: changed from `no` to `yes`
+  ac_cv_func_malloc_0_nonnull=yes
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2013-10-14 21:38:10.000000000 +0000
+++ src/setup.py	2018-12-29 16:05:22.612534092 +0000
@@ -272,6 +272,11 @@
             if os.system("chmod 0755 configure") != 0:
                 raise RuntimeError("chmod error")
             cmd = "sh configure"    # we use "sh" here so that it'll work on mingw32 with standard python.org binaries
+
+            # Chaquopy
+            host_triplet = os.path.basename(os.environ["CC"]).replace("-gcc", "")
+            cmd += " --host=" + host_triplet
+
             if self.verbose < 1:
                 cmd += " -q"
             if os.system(cmd) != 0:
@@ -353,16 +358,7 @@
                   "Crypto.Random",
                   "Crypto.Random.Fortuna",
                   "Crypto.Random.OSRNG",
-                  "Crypto.SelfTest",
-                  "Crypto.SelfTest.Cipher",
-                  "Crypto.SelfTest.Hash",
-                  "Crypto.SelfTest.Protocol",
-                  "Crypto.SelfTest.PublicKey",
-                  "Crypto.SelfTest.Random",
-                  "Crypto.SelfTest.Random.Fortuna",
-                  "Crypto.SelfTest.Random.OSRNG",
-                  "Crypto.SelfTest.Util",
-                  "Crypto.SelfTest.Signature",
+                  # Chaquopy: removed large package Crypto.SelfTest
                   "Crypto.Protocol",
                   "Crypto.PublicKey",
                   "Crypto.Signature"],
