--- src-original/f_check	2017-07-24 04:03:35.000000000 +0000
+++ src/f_check	2018-04-16 12:25:08.347006310 +0000
@@ -71,7 +71,10 @@
 
 	if ($data =~ /GNU/) {
 
-	    $data =~ /(\d)\.(\d).(\d)/;
+            # Chaquopy: Android GCC version number only has 2 components. I'm not
+            # going to try fixing this because I don't know what might be
+            # depending on it (see history of toolchain/gcc/gcc-4.9/gcc/BASE-VER).
+            $data =~ /(\d)\.(\d)\.x/;
 	    $major = $1;
 	    $minor = $2;
 
