diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2018-08-26 13:00:28.012674587 +0000
+++ src/setup.py	2018-08-26 13:52:26.893666837 +0000
@@ -36,6 +36,9 @@
         cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
                       '-DPYTHON_EXECUTABLE=' + sys.executable]
 
+        # Chaquopy added
+        cmake_args += ["-DCMAKE_TOOLCHAIN_FILE=" + os.path.abspath("../chaquopy.toolchain.cmake")]
+
         cfg = 'Debug' if self.debug else 'Release'
         build_args = ['--config', cfg]
 
