--- src-original/setup.py	2019-09-10 10:19:01.000000000 +0000
+++ src/setup.py	2020-01-27 16:00:07.729861962 +0000
@@ -93,7 +93,7 @@
         with open(os.path.join(root, "README.md")) as f:
             readme = f.read()
 
-        include_dirs = [get_python_inc(plat_specific=True)]
+        include_dirs = []  # Chaquopy: was [get_python_inc(plat_specific=True)]
 
         if (
             ccompiler.new_compiler().compiler_type == "msvc"
