--- src-original/src/_argon2_cffi_bindings/_ffi_build.py	2020-05-10 14:00:25.000000000 +0000
+++ src/src/_argon2_cffi_bindings/_ffi_build.py	2021-02-27 16:03:58.134283521 +0000
@@ -38,7 +38,7 @@
 else:
     # Optimized version requires SSE2 extensions. They have been around since
     # 2001 so we try to compile it on every recent-ish x86.
-    optimized = target_platform in ("i686", "x86", "x86_64", "AMD64")
+    optimized = any(word in os.environ["HOST"] for word in ("i686", "x86", "x86_64", "AMD64"))  # Chaquopy
 
 
 ffi = FFI()
