--- src-original/setup.py     2022-05-24 19:43:02.000000000 +0000
+++ src/setup.py        2023-04-06 16:08:59.631926941 +0000
@@ -1,6 +1,10 @@
 #!/usr/bin/env python
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+# Chaquopy
+import builtins
+builtins.__NUMPY_SETUP__ = True  # Prevent the compiled parts from being imported.
+
 from setuptools import setup
 
 from extension_helpers import get_extensions
