diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2019-03-08 12:49:03.000000000 +0000
+++ src/setup.py	2019-04-03 12:01:36.827146162 +0000
@@ -140,6 +140,20 @@
         cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
                       '-DPYTHON_EXECUTABLE=' + sys.executable]
 
+        # Chaquopy
+        host = os.environ["HOST"]
+        cmake_args += [
+            "-DCMAKE_VERBOSE_MAKEFILE=TRUE",
+            "-DDLIB_USE_CUDA=no",
+
+            # By default, dlib/cmake_utils/set_compiler_specific_options.cmake sets these
+            # options by running executables on the host, so force them according to
+            # https://developer.android.com/ndk/guides/abis.
+            f"-DSSE4_IS_AVAILABLE_ON_HOST={host.startswith('x86_64')}",
+            f"-DAVX_IS_AVAILABLE_ON_HOST=False",
+            f"-DARM_NEON_IS_AVAILABLE={host.startswith('arm')}",
+        ]
+
         cmake_args += cmake_extra_options 
 
         cfg = 'Debug' if self.debug else 'Release'
