--- src-original/setup.py	2025-05-22 12:01:25.000000000 +0000
+++ src/setup.py	2025-11-06 11:09:33.691098552 +0000
@@ -1,6 +1,11 @@
 #!/usr/bin/env python
 from setuptools import Extension, setup, find_packages
 import sys
+
+# Chaquopy: prevent the compiled parts of NumPy from being imported.
+import builtins
+builtins.__NUMPY_SETUP__ = True
+
 import numpy
 from setuptools.command.build_ext import build_ext
 from sysconfig import get_path
