diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py       2018-07-21 13:58:06.000000000 +0000
+++ src/setup.py        2018-08-12 19:28:32.242009293 +0000
@@ -143,9 +143,16 @@
         with io.open(os.path.join(root, 'README.rst'), encoding='utf8') as f:
             readme = f.read()

+        # Chaquopy: remove bundled Numpy headers: they are for x86-64. Use our own instead.
+        from shutil import rmtree
+        bundled_include = os.path.join(root, 'include')
+        bundled_numpy = os.path.join(bundled_include, 'numpy')
+        if os.path.exists(bundled_numpy):
+            rmtree(bundled_numpy)
         include_dirs = [
             get_python_inc(plat_specific=True),
-            os.path.join(root, 'include')]
+            bundled_include,
+            os.path.abspath("../requirements/numpy/core/include")]

         if (ccompiler.new_compiler().compiler_type == 'msvc'
         and msvccompiler.get_build_version() == 9):
