--- Python-3.12.0-original/configure	2023-11-22 09:33:49
+++ Python-3.12.0/configure	2023-11-22 10:13:05
@@ -7476,6 +7476,7 @@
   case $ac_sys_system in
     CYGWIN*)
           LDLIBRARY='libpython$(LDVERSION).dll.a'
+          BLDLIBRARY='-L. -lpython$(LDVERSION)'
           DLLLIBRARY='libpython$(LDVERSION).dll'
           ;;
     SunOS*)
@@ -24374,7 +24375,7 @@
 # On Android and Cygwin the shared libraries must be linked with libpython.
 
 if test "$PY_ENABLE_SHARED" = "1" && ( test -n "$ANDROID_API_LEVEL" || test "$MACHDEP" = "cygwin"); then
-  LIBPYTHON="-lpython${VERSION}${ABIFLAGS}"
+  LIBPYTHON="$BLDLIBRARY"
 else
   LIBPYTHON=''
 fi
--- Python-3.12.0-original/Modules/makesetup	2023-10-02 12:48:14
+++ Python-3.12.0/Modules/makesetup	2023-11-22 10:11:40
@@ -86,18 +86,6 @@
 # Newline for sed i and a commands
 NL='\
 '
-
-# Setup to link with extra libraries when making shared extensions.
-# Currently, only Cygwin needs this baggage.
-case `uname -s` in
-CYGWIN*) if test $libdir = .
-	 then
-	 	ExtraLibDir=.
-	 else
-	 	ExtraLibDir='$(LIBPL)'
-	 fi
-	 ExtraLibs="-L$ExtraLibDir -lpython\$(LDVERSION)";;
-esac
 
 # Main loop
 for i in ${*-Setup}
@@ -286,7 +274,7 @@
 				;;
 			esac
 			rule="$file: $objs"
-			rule="$rule; \$(BLDSHARED) $objs $libs $ExtraLibs -o $file"
+			rule="$rule; \$(BLDSHARED) $objs $libs \$(LIBPYTHON) -o $file"
 			echo "$rule" >>$rulesf
 		done
 	done
