--- a/configure
+++ b/configure
@@ -8373,7 +8373,12 @@ $as_echo "#define HAVE_HTOLE64 1" >>confdefs.h

 fi

-use_lfs=yes
+# Chaquopy: changed "yes" to "no". _LARGEFILE_SOURCE has no effect on Android, and
+# _FILE_OFFSET_BITS=64 has no effect on 64-bit ABIs, but on 32-bit ABIs it causes many critical
+# functions to disappear on API levels older than 24. See
+# https://android.googlesource.com/platform/bionic/+/master/docs/32-bit-abi.md.
+use_lfs=no
+
 # Don't use largefile support for GNU/Hurd
 case $ac_sys_system in GNU*)
   use_lfs=no
