--- src-original/c/malloc_closure.h     2019-11-03 14:22:31.000000000 +0000
+++ src/c/malloc_closure.h      2020-01-25 10:03:58.602896812 +0000
@@ -42,6 +42,8 @@
         return 0;
     ret = 0;
 
+    /* Chaquopy: getline requires API level 18, but PaX isn't used by Android anyway. */
+    #if __ANDROID_API__ >= 18
     while (getline (&buf, &len, f) != -1)
         if (!strncmp (buf, "PaX:", 4))
             {
@@ -51,6 +53,8 @@
                 break;
             }
     free (buf);
+    #endif
+
     fclose (f);
     return ret;
 }
diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2018-02-27 18:15:41.000000000 +0000
+++ src/setup.py	2018-05-08 23:01:46.812824736 +0000
@@ -9,8 +9,7 @@
 
 sources = ['c/_cffi_backend.c']
 libraries = ['ffi']
-include_dirs = ['/usr/include/ffi',
-                '/usr/include/libffi']    # may be changed by pkg-config
+include_dirs = []    # Chaquopy: removed /usr/include paths
 define_macros = []
 library_dirs = []
 extra_compile_args = []
@@ -138,7 +137,7 @@
     sources.extend(os.path.join(COMPILE_LIBFFI, filename)
                    for filename in _filenames)
 else:
-    use_pkg_config()
+    # use_pkg_config()  # Chaquopy: disabled
     ask_supports_thread()
     ask_supports_sync_synchronize()
 
