diff -urN src-original/obspy/core/util/libnames.py src/obspy/core/util/libnames.py
--- src-original/obspy/core/util/libnames.py	2017-10-27 20:55:42.000000000 +0000
+++ src/obspy/core/util/libnames.py	2018-08-20 20:02:57.311257055 +0000
@@ -51,8 +51,9 @@
         file we have to add this suffix, but not during building.
     """
     # our custom defined part of the extension file name
-    libname = "lib%s_%s_%s_py%s" % (
-        lib, platform.system(), platform.architecture()[0],
+    # Chaquopy: hard-code this to the build platform "Linux_64bit".
+    libname = "lib%s_Linux_64bit_py%s" % (
+        lib,  # platform.system(), platform.architecture()[0],
         ''.join([str(i) for i in platform.python_version_tuple()[:2]]))
     libname = cleanse_pymodule_filename(libname)
     # NumPy distutils adds extension suffix by itself during build (#771, #755)
