diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py	2017-10-27 20:55:44.000000000 +0000
+++ src/setup.py	2018-08-20 19:13:34.700317201 +0000
@@ -29,6 +29,16 @@
 except ImportError:
     pass

+# Chaquopy added
+import os
+import sys
+if sys.version_info[0] < 3:
+    import __builtin__ as builtins
+else:
+    import builtins
+sys.path.append(os.path.abspath("../requirements"))  # For numpy.distutils
+builtins.__NUMPY_SETUP__ = True  # Prevent the rest of NumPy from being imported.
+
 try:
     import numpy  # @UnusedImport # NOQA
 except ImportError:
