--- src-original/opencv/cmake/OpenCVDetectPython.cmake  2021-01-02 12:14:24.000000000 +0000
+++ src/opencv/cmake/OpenCVDetectPython.cmake   2021-01-27 16:40:13.164104124 +0000
@@ -123,7 +123,7 @@
   if(_found)
     set(_version_major_minor "${_version_major}.${_version_minor}")

-    if(NOT ANDROID AND NOT APPLE_FRAMEWORK)
+    if(NOT APPLE_FRAMEWORK)  # Chaquopy: removed ANDROID exclusion
       ocv_check_environment_variables(${library_env} ${include_dir_env})
       if(NOT ${${library_env}} STREQUAL "")
           set(PYTHON_LIBRARY "${${library_env}}")
@@ -144,8 +144,8 @@
         # Copy outputs
         set(_libs_found ${PYTHONLIBS_FOUND})
         set(_libraries ${PYTHON_LIBRARIES})
-        set(_include_path ${PYTHON_INCLUDE_PATH})
-        set(_include_dirs ${PYTHON_INCLUDE_DIRS})
+        set(_include_path ${PYTHON_INCLUDE_DIR})  # Chaquopy: replaced PYTHON_INCLUDE_PATH and
+        set(_include_dirs ${PYTHON_INCLUDE_DIR})  # PYTHON_INCLUDE_DIRS, which were always empty.
         set(_debug_libraries ${PYTHON_DEBUG_LIBRARIES})
         set(_libs_version_string ${PYTHONLIBS_VERSION_STRING})
         set(_debug_library ${PYTHON_DEBUG_LIBRARY})
@@ -175,7 +175,7 @@
       endif()
     endif()

-    if(NOT ANDROID AND NOT IOS AND NOT XROS)
+    if(NOT IOS AND NOT XROS)  # Chaquopy: removed ANDROID exclusion
       if(CMAKE_HOST_UNIX)
         execute_process(COMMAND ${_executable} -c "from distutils.sysconfig import *; print(get_python_lib())"
                         RESULT_VARIABLE _cvpy_process
@@ -240,7 +240,7 @@
                           OUTPUT_STRIP_TRAILING_WHITESPACE)
         endif()
       endif()
-    endif(NOT ANDROID AND NOT IOS AND NOT XROS)
+    endif(NOT IOS AND NOT XROS) # Chaquopy: removed ANDROID exclusion
   endif()
 
   # Export return values
--- src-org/opencv/CMakeLists.txt  2021-01-02 12:14:24.000000000 +0000
+++ src/opencv/CMakeLists.txt   2021-01-27 15:35:52.366083600 +0000
@@ -428,7 +428,7 @@
   VISIBLE_IF UNIX AND NOT ANDROID AND NOT IOS
   VERIFY HAVE_GPHOTO2)
 OCV_OPTION(WITH_LAPACK "Include Lapack library support" (NOT CV_DISABLE_OPTIMIZATION)
-  VISIBLE_IF NOT ANDROID AND NOT IOS AND NOT XROS
+  VISIBLE_IF NOT IOS AND NOT XROS # Chaquopy: removed ANDROID exclusion
   VERIFY HAVE_LAPACK)
 OCV_OPTION(WITH_ITT "Include Intel ITT support" ON
   VISIBLE_IF NOT APPLE_FRAMEWORK
--- src_org/opencv/modules/python/CMakeLists.txt
+++ src/opencv/modules/python/CMakeLists.txt
@@ -3,7 +3,7 @@
 # ----------------------------------------------------------------------------
 if(DEFINED OPENCV_INITIAL_PASS)  # OpenCV build
 
-if(ANDROID OR APPLE_FRAMEWORK OR WINRT)
+if(APPLE_FRAMEWORK OR WINRT) # Chaquopy: removed ANDROID exclusion
   ocv_module_disable_(python2)
   ocv_module_disable_(python3)
   return()
--- src_org/setup.py	2025-01-16 17:51:31.000000000 +0800
+++ src/setup.py	2025-03-25 11:42:40.058255000 +0800
@@ -17,8 +17,11 @@
     is_CI_build = True if CI_BUILD == "1" else False
     cmake_source_dir = "opencv"
     minimum_supported_numpy = "1.13.3"
-    build_contrib = get_build_env_var_by_name("contrib")
-    build_headless = get_build_env_var_by_name("headless")
+
+    # Chaquopy
+    build_contrib = "contrib" in os.environ["PKG_NAME"]
+    build_headless = "headless" in os.environ["PKG_NAME"]
+
     build_java = "ON" if get_build_env_var_by_name("java") else "OFF"
     build_rolling = get_build_env_var_by_name("rolling")
 
@@ -36,17 +39,26 @@
     ]
 
     python_version = cmaker.CMaker.get_python_version()
-    python_lib_path = cmaker.CMaker.get_python_library(python_version) or ""
-    # HACK: For Scikit-build 0.17.3 and newer that returns None or empty sptring for PYTHON_LIBRARY in manylinux2014
-    # A small release related to PYTHON_LIBRARY handling changes in 0.17.2; scikit-build 0.17.3 returns an empty string from get_python_library if no Python library is present (like on manylinux), where 0.17.2 returned None, and previous versions returned a non-existent path. Note that adding REQUIRED to find_package(PythonLibs will fail, but it is incorrect (you must not link to libPython.so) and was really just injecting a non-existent path before.
-    # TODO: Remove the hack when the issue is handled correctly in main OpenCV CMake.
-    if python_lib_path == "":
-        python_lib_path = "libpython%sm.a" % python_version
-    python_lib_path = python_lib_path.replace("\\", "/")
 
-    python_include_dir = cmaker.CMaker.get_python_include_dir(python_version).replace(
-        "\\", "/"
-    )
+    # Chaquopy
+    python_include_dir = None
+    in_dirafter = False
+    for word in os.environ["CFLAGS"].split():
+        if word == "-idirafter":
+            in_dirafter = True
+        elif in_dirafter:
+            in_dirafter = False
+            if "include/python" in word:
+                python_include_dir = word
+    assert python_include_dir, os.environ["CFLAGS"]
+
+    # Chaquopy
+    python_lib_dir = os.path.abspath(python_include_dir + "/../../lib")
+    python_lib_path = None
+    for word in os.environ["LDFLAGS"].split():
+        if word.startswith("-lpython"):
+            python_lib_path = f"{python_lib_dir}/lib{word[2:]}.so"
+    assert python_lib_path, os.environ["LDFLAGS"]
 
     if os.path.exists(".git"):
         import pip._internal.vcs.git as git
@@ -79,7 +91,7 @@
     )
 
     # https://stackoverflow.com/questions/1405913/python-32bit-or-64bit-mode
-    is64 = sys.maxsize > 2 ** 32
+    is64 = "64" in os.environ["CHAQUOPY_ABI"]  # Chaquopy: was sys.maxsize > 2 ** 32
 
     package_name = "opencv-python"
 
@@ -95,6 +107,9 @@
     if build_rolling:
         package_name += "-rolling"
 
+    # Chaquopy
+    package_name = os.environ["PKG_NAME"]
+
     long_description = io.open("README.md", encoding="utf-8").read()
 
     packages = ["cv2", "cv2.data"]
@@ -137,7 +152,8 @@
         [ r"python/cv2/py.typed" ] if sys.version_info >= (3, 6) else []
         ,
         "cv2.data": [  # OPENCV_OTHER_INSTALL_PATH
-            ("etc" if os.name == "nt" else "share/opencv4") + r"/haarcascades/.*\.xml"
+            # Chaquopy: see opencv/cmake/OpenCVInstallLayout.cmake
+            "sdk/etc" + r"/haarcascades/.*\.xml"
         ],
         "cv2.gapi": [
             "python/cv2" + r"/gapi/.*\.py"
@@ -213,6 +229,31 @@
         )
     )
 
+    # Chaquopy
+    py_major_ver = sys.version_info[0]
+    cmake_args += [
+        # Enable CMake's cross-compilation mode.
+        "-DCMAKE_TOOLCHAIN_FILE=" + os.path.abspath("../chaquopy.toolchain.cmake"),
+
+        # Unnecessary, and requires an old version of the SDK.
+        "-DBUILD_ANDROID_PROJECTS=OFF", "-DBUILD_ANDROID_EXAMPLES=OFF",
+
+        # External dependencies which we've built as wheels (listed in meta.yaml).
+        "-DBUILD_JPEG=OFF", "-DBUILD_PNG=OFF",
+
+        # External dependencies which we can do without for now.
+        "-DWITH_TIFF=OFF", "-DWITH_JASPER=OFF", "-DWITH_OPENEXR=OFF", "-DWITH_WEBP=OFF",
+
+        # These are only available for x86, so disable for consistency between ABIs.
+        "-DWITH_IPP=OFF", "-DWITH_ITT=OFF",
+
+        # Python native modules aren't usually linked against libpython on Linux, but our
+        # configuration requires it (see opencv/modules/python/common.cmake).
+        "-DOPENCV_FORCE_PYTHON_LIBS=ON",
+        "-DPYTHON{}_NUMPY_INCLUDE_DIRS={}".format(
+            py_major_ver, os.path.abspath("../requirements/numpy/core/include")),
+    ]
+
     if build_headless:
         # it seems that cocoa cannot be disabled so on macOS the package is not truly headless
         cmake_args.append("-DWITH_WIN32UI=OFF")
--- src_org/opencv/modules/python/common.cmake	2025-01-16 17:51:31.000000000 +0800
+++ src/opencv/modules/python/common.cmake	2025-03-25 11:42:40.058255000 +0800
@@ -14,7 +14,8 @@
 
 # try to use dynamic symbols linking with libpython.so
 set(OPENCV_FORCE_PYTHON_LIBS OFF CACHE BOOL "")
+ string(REPLACE "-Wl,--no-undefined-version" "" CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS}")  # Chaquopy: fixbug for '-version' while compiling
 string(REPLACE "-Wl,--no-undefined" "" CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS}")
 if(NOT WIN32 AND NOT APPLE AND NOT OPENCV_PYTHON_SKIP_LINKER_EXCLUDE_LIBS)
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--exclude-libs=ALL")
