--- src-original/_setuplibev.py 2023-09-12 17:14:39.000000000 +0000
+++ src/_setuplibev.py  2023-11-09 16:39:57.623434764 +0000
@@ -25,10 +25,15 @@
 
 LIBEV_EMBED = should_embed('libev')
 
+# Chaquopy
+import os
+host = os.environ.get("HOST")
+host_arg = f" --host={host}" if host else ""
+
 # Configure libev in place
 libev_configure_command = ' '.join([
     "(cd ", quoted_dep_abspath('libev'),
-    " && sh ./configure -C > configure-output.txt",
+    f" && sh ./configure {host_arg} -C > configure-output.txt",
     ")",
 ])
 
diff -ur src-original/setup.py src/setup.py
--- src-original/setup.py       2019-01-04 11:51:44.000000000 +0000
+++ src/setup.py        2019-08-05 20:46:00.204601700 +0000
@@ -160,7 +165,8 @@
 
 EXT_MODULES = [
     CORE,
-    ARES,
+    # Chaquopy: disable ARES: it has build errors, and it isn't used by default anyway.
+    # ARES,
     ABSTRACT_LINKABLE,
     SEMAPHORE,
     LOCAL,
--- src-original/src/gevent/libuv/_corecffi_build.py    2023-09-12 17:14:39.000000000 +0000
+++ src/src/gevent/libuv/_corecffi_build.py     2023-11-09 17:05:14.935975179 +0000
@@ -155,6 +155,9 @@
         _libuv_source('unix/thread.c'),
         _libuv_source('unix/tty.c'),
         _libuv_source('unix/udp.c'),
+
+        # Chaquopy
+        _libuv_source('unix/pthread-fixes.c'),
     ]
 
 
