--- src-original/setup.py	2019-10-19 08:15:34.000000000 +0000
+++ src/setup.py	2022-10-04 21:19:55.423152993 +0000
@@ -175,10 +175,14 @@
 # C files must be built once only for coverage to work
 c_lib = ('c_wt', {'sources': sources,
                   'depends': source_templates + header_templates + headers,
-                  'include_dirs': [make_ext_path("c"), get_python_inc()],
+                  'include_dirs': [make_ext_path("c")],  # Chaquopy: removed get_python_inc()
                   'macros': c_macros, })
 
 def get_ext_modules(use_cython):
+    # Chaquopy
+    import builtins
+    builtins.__NUMPY_SETUP__ = True  # Prevent the compiled parts from being imported.
+
     from numpy import get_include as get_numpy_include
     cython_modules, cython_sources = get_cython_sources(use_cython)
     ext_modules = [
@@ -439,7 +443,7 @@
         tests_require=['pytest'],
 
         install_requires=["numpy>=1.13.3"],
-        setup_requires=["numpy>=1.13.3"],
+        # Chaquopy: removed setup_requires=["numpy>=1.13.3"],
         python_requires=">=3.5",
     )
 
