--- src-original/setup.py	2025-08-12 05:52:58.000000000 +0000
+++ src/setup.py	2025-11-05 19:33:33.331371791 +0000
@@ -227,6 +227,7 @@
   SENTENCEPIECE_EXT = Extension(
       'sentencepiece._sentencepiece',
       sources=['src/sentencepiece/sentencepiece_wrap.cxx'],
+      libraries=["log"],  # Chaquopy: see bundled.patch
   )
   cmdclass = {'build_ext': build_ext}
 
--- src-original/build_bundled.sh	2025-08-12 05:52:58.000000000 +0000
+++ src/build_bundled.sh	2025-11-05 19:36:35.233764329 +0000
@@ -18,5 +18,8 @@
   SRC_DIR=./sentencepiece
 fi
 
+# Chaquopy
+patch -d $SRC_DIR -p1 -i $RECIPE_DIR/bundled.patch || exit 1
+
 cmake ${SRC_DIR} -B ${BUILD_DIR} -DSPM_ENABLE_SHARED=OFF -DSPM_DISABLE_EMBEDDED_DATA=ON -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}
 cmake --build ${BUILD_DIR} --config Release --target install --parallel $(nproc)
