--- src-original/setup.py	2021-01-08 16:09:18.000000000 +0000
+++ src/setup.py	2021-01-11 22:03:33.904597583 +0000
@@ -98,7 +98,8 @@
 else:
   SENTENCEPIECE_EXT = Extension(
       'sentencepiece._sentencepiece',
-      sources=['src/sentencepiece/sentencepiece_wrap.cxx'])
+      sources=['src/sentencepiece/sentencepiece_wrap.cxx'],
+      libraries=["log"])  # Chaquopy: added
   cmdclass = {'build_ext': build_ext}

 setup(
--- src-original/build_bundled.sh	2021-01-09 15:24:24.000000000 +0000
+++ src/build_bundled.sh	2023-12-16 22:18:22.971159515 +0000
@@ -4,12 +4,12 @@
 
 mkdir bundled
 cd bundled
-# Try taged version. Othewise, use head.
+# Chaquopy: removed fallback onto HEAD.
 git clone https://github.com/google/sentencepiece.git \
-  -b v"${VERSION}" --depth 1 || \
-  git clone https://github.com/google/sentencepiece.git --depth 1
+  -b v"${VERSION}" --depth 1
 
 cd sentencepiece
+patch -p1 -i $RECIPE_DIR/bundled.patch  # Chaquopy
 mkdir build
 cd build
 cmake .. -DSPM_ENABLE_SHARED=OFF -DCMAKE_INSTALL_PREFIX=../..
