--- src-original/setup.py	2021-01-08 16:09:18.000000000 +0000
+++ src/setup.py	2021-01-11 22:03:33.904597583 +0000
@@ -98,7 +98,8 @@
 else:
   SENTENCEPIECE_EXT = Extension(
       'sentencepiece._sentencepiece',
-      sources=['src/sentencepiece/sentencepiece_wrap.cxx'])
+      sources=['src/sentencepiece/sentencepiece_wrap.cxx'],
+      libraries=["log"])  # Chaquopy: added
   cmdclass = {'build_ext': build_ext}

 setup(
--- src-original/build_bundled.sh	2021-01-09 15:24:24.000000000 +0000
+++ src/build_bundled.sh	2021-01-12 13:35:07.595453948 +0000
@@ -1,18 +1,20 @@
 #!/bin/sh
 
 VERSION="$1"
+export LDFLAGS="$LDFLAGS -llog"  # Chaquopy
 
 mkdir bundled
 cd bundled
-# Try taged version. Othewise, use head.
+# Chaquopy: removed fallback onto HEAD.
 git clone https://github.com/google/sentencepiece.git \
-  -b v"${VERSION}" --depth 1 || \
-  git clone https://github.com/google/sentencepiece.git --depth 1
+  -b v"${VERSION}" --depth 1
 
 cd sentencepiece
+patch -p1 -i $SRC_DIR/util.patch  # Chaquopy: fix missing #include.
 mkdir build
 cd build
-cmake .. -DSPM_ENABLE_SHARED=OFF -DCMAKE_INSTALL_PREFIX=../..
+cmake .. -DSPM_ENABLE_SHARED=OFF -DCMAKE_INSTALL_PREFIX=../.. \
+      -DCMAKE_TOOLCHAIN_FILE=$SRC_DIR/../chaquopy.toolchain.cmake  # Chaquopy
 make -j $(nproc)
 make install
 cd ../..
--- src-original/util.patch	1970-01-01 00:00:00.000000000 +0000
+++ src/util.patch	2021-01-12 13:30:53.587344655 +0000
@@ -0,0 +1,18 @@
+diff --git a/src/util.h b/src/util.h
+index 18d6e9c..08c2e00 100644
+--- a/src/util.h
++++ b/src/util.h
+@@ -36,9 +36,12 @@
+ #include <pthread.h>
+ #endif
+ 
+-#if !defined(__APPLE__) && !defined(_WIN32) && BYTE_ORDER == __BIG_ENDIAN
++#if !defined(__APPLE__) && !defined(_WIN32)
++#include <endian.h>
++#if defined(BYTE_ORDER) && defined(__BIG_ENDIAN) && BYTE_ORDER == __BIG_ENDIAN
+ #define IS_BIG_ENDIAN
+ #endif
++#endif
+ 
+ namespace sentencepiece {
+ template <typename T>
