diff -ru flac-1.3.3-orig/config.sub flac-1.3.3/config.sub
--- flac-1.3.3-orig/config.sub	2018-02-25 00:00:57.000000000 +0800
+++ flac-1.3.3/config.sub	2022-09-17 10:22:55.000000000 +0800
@@ -116,6 +116,7 @@
 case $maybe_os in
   nto-qnx* | linux-gnu* | linux-android* | linux-dietlibc | linux-newlib* | \
   linux-musl* | linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | \
+  ios*-simulator | tvos*-simulator | watchos*-simulator | \
   knetbsd*-gnu* | netbsd*-gnu* | netbsd*-eabi* | \
   kopensolaris*-gnu* | cloudabi*-eabi* | \
   storm-chaos* | os2-emx* | rtmk-nova*)
@@ -249,7 +250,7 @@
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
 	| arc | arceb \
-	| arm | arm[bl]e | arme[lb] | armv[2-8] | armv[3-8][lb] | armv7[arm] \
+	| arm | arm[bl]e | arme[lb] | armv[2-8] | armv[3-8][lb] | armv7[armk] \
 	| avr | avr32 \
 	| ba \
 	| be32 | be64 \
@@ -450,6 +451,10 @@
 	| ymp-* \
 	| z8k-* | z80-*)
 		;;
+	arm64-* | arm64e-* | arm64_32-*)
+ 		basic_machine=aarch64-`echo "$basic_machine" | sed 's/^[^-]*-//'`
+ 		;;
+
 	# Recognize the basic CPU types without company name, with glob match.
 	xtensa*)
 		basic_machine=$basic_machine-unknown
@@ -1520,8 +1525,12 @@
 		;;
 	-nacl*)
 		;;
-	-ios)
+	-ios | -ios-simulator)
+		;;
+	-tvos | -tvos-simulator)
 		;;
+	-watchos | -watchos-simulator)
+ 		;;
 	-none)
 		;;
 	*)
