--- src-original/setup.py	2019-11-13 12:12:11.000000000 +0000
+++ src/setup.py	2020-01-25 12:44:02.672871158 +0000
@@ -93,7 +93,7 @@
         with io.open(os.path.join(root, "README.md"), encoding="utf8") as f:
             readme = f.read()
 
-        include_dirs = [get_python_inc(plat_specific=True)]
+        include_dirs = []  # Chaquopy: removed [get_python_inc(plat_specific=True)]
 
         ext_modules = []
         for mod_name in MOD_NAMES:
